pkg/cmd/issue/create/create.go

```diff
@@ -140,24 +152,28 @@ func createRun(opts *CreateOptions) error {
 
 	repo, err := api.GitHubRepo(apiClient, baseRepo)
 	if err != nil {
-		return err
+		return
 	}
 	if !repo.HasIssuesEnabled {
-		return fmt.Errorf("the '%s' repository has disabled issues", ghrepo.FullName(baseRepo))
+		err = fmt.Errorf("the '%s' repository has disabled issues", ghrepo.FullName(baseRepo))
+		return
 	}
 
 	action := prShared.SubmitAction
 
 	if opts.Interactive {
-		editorCommand, err := cmdutil.DetermineEditor(opts.Config)
+		var editorCommand string
+		editorCommand, err = cmdutil.DetermineEditor(opts.Config)
 		if err != nil {
-			return err
+			return
 		}
 
+		defer prShared.PreserveInput(opts.IO, &tb, &err)()
```

Comment:
I wondered why you named the `(err error)` return value in the parent function, and now I understand. Clever solution for capturing any error from this point onward!

Could preserving input on any error be over-zealous, however? I can imagine some instances where an error being returned from `createRun()` doesn't necessarily have to result in restore instructions being printed to stderr. For instance, the user might have explicitly aborted editing in their text editor (e.g. `:cq` in vim), or pressed <kbd>Ctrl-C</kbd> to abort the interactive flow. Since the user chose to bail out from the command, I don't think we need to preserve their input there.

Additionally, the defer approach will fail to preserve input if the program crashes while making the request or processing the HTTP response. (We don't know how often this happens, but it's worth keeping in mind.)

Would it be enough to simply preserve input just before dispatching the API command, and only handle the error that is the result of it? Sort of like (pseudocode):

```
if wasInteractive {
  PreserveInput(tb)
}
pr, err := createPR(data)
if err != nil {
  ShowRestoreInstructions()
}
```

Reply:
> For instance, the user might have explicitly aborted editing in their text editor (e.g. :cq in vim), or pressed Ctrl-C

I pretty strongly think that manually aborting _is_ an appropriate time to save input. I've personally realized that something is wrong with my branch or `create pr` invocation after drafting a body and am forced to either re-type the body or submit something I'm not ready to submit then clean it up.

> Additionally, the defer approach will fail to preserve input if the program crashes while making the request or processing the HTTP response. (We don't know how often this happens, but it's worth keeping in mind.)

While I am open to moving away from `defer`, just to be clear, this would be an appropriate usecase for `recover`, yes?

Reply:
If your intended goal is to capture any error after this point, not just the API error from the `createPullRequest` mutation, then I do agree that the `defer` approach is worth sticking with. 👍 

I would not advise using `recover` for most things other than crash/error reporting.

Reply:
My thinking with `recover` would be that we could just dump any user input to STDOUT in case of a panic but I think we can worry about that later if it actually bites users.